kit/src/elements/file/mod.rs

```diff
@@ -135,14 +145,43 @@ pub fn File<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                         }
                     },
                 },
-                with_rename.then(|| rsx! (
-                    Input {
-                        disabled: disabled,
-                        placeholder: placeholder,
-                        // todo: use is_valid
-                        onreturn: move |(s, _is_valid)| emit(&cx, s)
+                with_rename.then(||
+                    {
+                    let chars_to_remove = vec!['\\', '/'];
+                    let mut special_chars = SPECIAL_CHARS.to_vec();
```

Comment:
i know this filtering of `SPECIAL_CHARS` is needed for the input validation. I think it's better to  add a `chars_to_allow` field on the input box. in this case it would be `vec!['\\', '/']`. the use of `SPECIAL_CHARS` could be moved inside of the `Input` element. 

without doing this, the logic for filtering allowed characters and then passing them to `Input` will be duplicated a lot. i think the sooner this gets changed, the better.

Reply:
also (not the fault of this PR), could the input validation just filter out everything in `chars_to_allow` before checking if alpha numeric? using `SPECIAL_CHARS` seems  unnecessary.